### PR TITLE
Add SessionProvider class

### PR DESCRIPTION
**Pull request by epenet**
This new class provides a layer of abstraction for Kamereon session, and Kamereon no longer depends directly on Gigya.

Implements #67

### CODE REVIEW

**Comment by epenet** [comment]
Hi @mfonville,
Do you want to have a quick look at this?

**Comment by mfonville** [comment]
Thanks for the notice, but I'd need some time to review this, but lack the time at this very moment.
But in general abstracting the Kamereon session is a good idea imho.  Thanks for using my comment on PyZE for this new implementation.

File: src/renault_api/kamereon/enums.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+"""Kamereon enums."""
+from enum import Enum
+
+
+class EnergyCode(Enum):
+    """Enum for vehicleDetails energy code."""
+
+    ESSENCE = "ESS"
+    ELECTRIQUE = "ELEC"
```
Comment (Quentame) [inline src/renault_api/kamereon/enums.py:9]:
```suggestion
    GASOLINE = "ESS"
    ELECTRIC = "ELEC"
```

File: src/renault_api/kamereon/models.py [hunk 2]
```diff
@@ -26,7 +47,15 @@ class KamereonResponseError(BaseModel):
 
     def raise_for_error_code(self) -> None:
         """Raise exception from response error."""
-        raise KamereonResponseException(self.errorCode, self.get_error_details())
+        error_details = self.get_error_details()
+        for common_error in COMMON_ERRRORS:
+            if (
+                self.errorCode == common_error["errorCode"]
+                and error_details == common_error["error_details"]
```
Comment (Quentame) [inline src/renault_api/kamereon/models.py:54]: I think `error_details` should be trusted on the API side.

Checking the code is enough.

If one day the details changes, you code will not work.

File: src/renault_api/kamereon/models.py [hunk 3]
```diff
@@ -1,20 +1,41 @@
...
+        "errorCode": "err.tech.501",
+        "error_details": "This feature is not technically supported by this gateway",
+        "error_type": exceptions.NotSupportedException,
+    },
+    {
+        "errorCode": "err.func.wired.overloaded",
+        "error_details": "You have reached your quota limit",
+        "error_type": exceptions.QuotaLimitException,
+    },
+]
```
Comment (Quentame) [inline src/renault_api/kamereon/models.py:38]:
```suggestion
COMMON_ERRRORS: List[Dict[str, Any]] = [
    {
        "errorCode": "err.func.403",
        "error_type": exceptions.AccessDeniedException,
    },
    {
        "errorCode": "err.tech.500",
        "error_type": exceptions.InvalidUpstreamException,
    },
    {
        "errorCode": "err.tech.501",
        "error_type": exceptions.NotSupportedException,
    },
    {
        "errorCode": "err.func.wired.overloaded",
        "error_type": exceptions.QuotaLimitException,
    },
]
```

As previous comment

File: src/renault_api/kamereon/models.py [hunk 4]
```diff
@@ -110,57 +121,53 @@ class KamereonVehiclesDetailsGroup(BaseModel):
...
+            return None
         return self.model.label
 
+    def reports_charging_power_in_watts(self) -> bool:
+        """Return True if model reports chargingInstantaneousPower in watts."""
+        if self.get_model_code() in [
+            "X101VE",  # ZOE 40
+        ]:
+            return True
+        return False
```
Comment (Quentame) [inline src/renault_api/kamereon/models.py:169]: No way to get it from the API ?

**Review by Quentame — COMMENTED** [review]
Seems nice

File: src/renault_api/kamereon/models.py (hunk 2 above)
Comment (mfonville) [reply src/renault_api/kamereon/models.py:54]: Renault's servers sometimes give the same errorcode with a different error detail. But indeed not sure if this code should differentiate on this detail level. (Probably just better to only pass that detail string along when raising the exception)

File: src/renault_api/kamereon/enums.py (hunk 1 above)
Comment (epenet) [reply src/renault_api/kamereon/enums.py]: I have removed the enum, to keep just the Renault string.
I have added uses_electricity() / uses_fuel() methods instead.

File: src/renault_api/kamereon/models.py (hunk 3 above)
Comment (epenet) [reply src/renault_api/kamereon/models.py]: Addressed in 1e48d0038b3e2bece4ed476e080271f1caa0d36f

File: src/renault_api/kamereon/models.py (hunk 4 above)
Comment (epenet) [reply src/renault_api/kamereon/models.py:174]: Until we have some more samples, then this is the only way that I know of. There was a discussion on PyZE about it...
It seems that Zoe50 reports in kW, and I expect all newer models will also default to kW.

**Pull request merged**